WordPress/gutenberg
# Add keyboard navigation to the inserter.

Fixes #515. This PR adds keyboard navigation to the inserter. This introduces RxJS as a library, which I feel could be very useful for other aspects of this project. Accessibility review would be awesome @afercia.

WIP: Need to fix testing errors which appear to be from the build, the PR should work. Need to fix weird RxJS issue.  Also rebase.

**Open questions:**

Is it worth adding another dependency for RxJS?  I think for a heavily active user interface like the editor we will probably want to add something like RxJS to help manage some of the interaction going on within the editor.  This does however add more complexity for the project as RxJS is probably not widely used in the WordPress ecosystem compared to React and Redux.

**Testing instructions:**

Verify that after toggling an inserter menu open you can navigate through the block types buttons via keyboard shortcuts ( right arrow, down arrow, and tab move forward; left arrow, up arrow, and shift + tab move backward, through the list. )  As you cycle through the block types and the search field verify that the focus status is being properly updated.  Press the escape key for toggling the inserter.  When toggling the inserter the toggle button for that inserter should recieve focus.

**Feedback**

How can I improve my PRs?
